### DISCUSSION TITLE
Sega mark 3 adaptor

### DISCUSSION THREAD

**Discussion by Stanton54** [sanni/cartreader#674] [Need help]
![image](https://user-images.githubusercontent.com/120297611/210473612-eb8eef24-7299-4f4e-b282-0a43eabee8c0.jpeg)

I am getting an error message each time I try to dump this cartridge.

I go to SMS/GG/Mlll/sg-1000 then SG-1000 Raphnet but I always get the same error SMS/GG Header not found.

I have the right rom size of 32kb selected for this game.

I tried resoldering all the pins but nothing changed. I cannot find this game on my sd card in any section in the current zip file.

Is this cartridge possibly incompatible? It should work for Sg-3000 or SG-1000 series.

When I try to dump it gives me an unusable UNKNOWN file name SMS format.

Thanks so much!

**Comment by Stanton54** [comment]
The CRC32 also changes each time I dump even with the same parameters. The name of the game is champion golf.

**Comment by Ancyker** [comment]
You should ensure the pins and contacts are clean on the cartridge and the adapter. I take apart all cartridges that are new to me and clean the pins/contacts super well with contact cleaner before doing anything else with them. I've had issues with them not working in the reader but working fine in the actual console/hardware. Since doing this, I have not had that issue, except for one cartridge. 

For one cartridge (my first SNES SA1) I had, it still would not dump correctly even though I had cleaned the pins. I took it apart again, and while inspecting it more closely, I noticed it was slightly sticky -- not super sticky. It was tacky enough that if you pressed your finger onto it, it would stick to it for a brief second as you lifted it up. It was barely noticeable, which is why I missed it when I opened it to clean the contacts. I repeatedly sprayed and cleaned the entire PCB with contact cleaner until the tackiness was gone. After that, it started working fine. I don't have a working SNES right now, so I couldn't test if it was working with the original hardware, but the seller claimed it did. There was no sign of liquid damage or other obvious issues. It was just slightly tacky to the touch. So, if cleaning the contacts doesn't resolve the issue, try cleaning the entire PCB.

**Reply by Stanton54** [reply]
I have never thought of cleaning the slot too. I have been lucky in the build so far but I will try cleaning it and the PCB (why not) to see how it goes!

**Comment by PsyK0p4T** [comment]
Hi @Stanton54 

Got some good news for you : this game is dumpable fine. I did it already, with a markIII adapter aswell.

To answer to your questions : 

"I always get the same error SMS/GG Header not found." > totally normal. SG1000 games don't contain the "TMR SEGA" header, like most of SMS/GG games. We should maybe manage the SG1000 a bit differently and skip this check for this console in particular.

"I have the right rom size of 32kb selected for this game." > yup, good one. The game is 32k. Dunno if you got the v0 or v1 but both are 32k so this is fine.

"I cannot find this game on my sd card in any section in the current zip file." > is your setup up to date ? With latest builds you should have a "sg1000.txt" file at your sd card root, and this rom is in it. Expected crc32 is 868419b5 for the v0 rom.

"Is this cartridge possibly incompatible? " > really i don't think so, i dumped it myself.  Have you already managed to dump other SMS / MarkIII / SG1000 games with your setup ? Just to be sure that your adapter is fine.

But... since we're talking about SG1000 in this thread, note that the dumper is currently incompatible with the 40k roms (like Monaco GP v0 for example). For now as a quick test I only tried to force the dumping size to 40k instead of 32 (which is the current maximum size) but this isn't enough. At least for games under or egal to 32k, the dumper should work. Try to clean your cart pins again, brush them with a rubber, they really need to be perfectly clean or you'll get a different crc everytime.

**Reply by Stanton54** [reply]
I have gone and cleaned the pins quite extensively. (I was watching a movie)

I cleaned off the PCB as best as I could and even the pcb for the cartridge.

I did not have the SG 1000 database in my SD card. I was using a sticky note with the codes for three known good dumps lol . I have now uploaded it to my sd card so my sd card is up to date like the cart reader.

I am not sure if the cartridge is broken. At this time I don’t have anything else to test it with.  I will try ordering another cartridge and see if that works.

I don’t any other games in the Sega Mark three. Personally, I played at lot of Super Nintendo and sega genesis (Still do). I am more doing it as an enthusiast just because I think the project is cool and I want to follow along. I only have the one adapter, so it could also be that something is wrong with the slot too. When I ordered the PCB, I set the thickness to 1.2mm.

Hopefully the new one will work. The other seller is claiming the champion golf worked with the original hardware as well.

**Reply by PsyK0p4T** [reply]
Also, in addition to the pin cleaning work and pcb thickness, also make sure that the cartridge itself is well plugged into the adapter.

I mean, with the adapter I'm using for example, the female slot doesn't have the perfect spacing between pins. That's just a matter of millimeters, but I've to aim right when I plug a mark3 cartridge in it. If i'm not perfectly centered, the very right or left pin of the cart could not be in contact. Resulting a bad dectection and bad dump.

**Reply by Stanton54** [reply]
![image](https://user-images.githubusercontent.com/120297611/211383779-1759dc7c-a41a-4c88-8560-66810cc6ff40.jpeg)

I tried adjusting it to be centered.

I wonder if the pins are too used up for a good dump? I cannot think of anything else.

I know it is clean at least. If it wasn't for the lead I think I could eat off it.

Do I need to enable an extension? Right now I just have the default settings for V10.2

**Reply by PsyK0p4T** [reply]
Hard to judge with just a picture, but I don't see it used so much. Have you tried to brush the pins with a soft rubber ?

The only important line you need in the config is "#define enable_SMS". But you have it enabled already, otherwise it would be impossible to access the SMS module and make a dump. About switches, make sure you are using 5V, and all others on left position (off).

Also, with your pictures I just noticed a little difference between your MARKIII adapter and mine : I got resistors soldered on it, not yours. https://user-images.githubusercontent.com/87064902/198355949-d04e7476-b531-4197-82a5-2dff7c2222af.jpg but I can't say if it's a problem or not. When you bought it, was it ready-to-use ?
Maybe try to grab other carts for more testing, MARKIII if possible, as the adapter was mainly designed for this console. SG-1000/3000 is a bonus.

**Reply by Stanton54** [reply]
I have another cart coming in to test and I am using the zip file from Ryhurron. I may have to add resistors. I did not see any completed ones so it is definitely possible I am missing them.

**Reply by Stanton54** [reply]
That line is enabled for sure as I dumped me already. I tried cleaning with an eraser and alcohol but that didn't help. It is definitely the cleanest cart I have now. I was worried about removing the gold plating so I stopped cleaning for now. I have a v5 so all of that should be set automatically and I do have volatage set to 5 volts. Do you know what resistors you have?

**Reply by PsyK0p4T** [reply]
No sorry, I looked at it but can't know their exact values. Also checked my SMS adapter, and this one doesn't use any resistors. So maybe it's not mandatory... more tests on different cartridges will tell.

**Reply by Stanton54** [reply]
Thanks! I will give that a shot
